packages/plugin-cosmos/src/actions/transfer.ts

```diff
@@ -0,0 +1,280 @@
+    }
+
+    const rpcEndpoint = chain.apis.rpc?.[0];
+    if (!rpcEndpoint) {
+        throw new Error(`RPC endpoint not found for chain ${chainName}`);
+    }
+
+    return {
+        batch32prefix,
+        feeToken,
+        rpcEndpoint,
+        chainAssets: assetList,
+    };
+};
+
+const createWallet = async (recoveryPhrase: string, batch32prefix: string) => {
+    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(recoveryPhrase, {
+        prefix: batch32prefix,
+    });
+
+    return wallet;
+};
+
+const createSigningClient = async (
+    rpcEndpoint: string,
+    wallet: DirectSecp256k1HdWallet,
+    feeTokenDenom: string
+) => {
+    const gasPrice = GasPrice.fromString(`0.025${feeTokenDenom}`);
+
+    const signingCosmWasmClient = await SigningCosmWasmClient.connectWithSigner(
+        rpcEndpoint,
+        wallet,
+        {
+            gasPrice: gasPrice,
+        }
+    );
+
+    return signingCosmWasmClient;
+};
+
+async function sendTokens(
+    signingCosmWasmClient: SigningCosmWasmClient,
+    senderAddress: string,
+    recipientAddress: string,
+    coin: Coin
+) {
+    const result = await signingCosmWasmClient.sendTokens(
+        senderAddress, // Sender address
+        recipientAddress, // Recipient address
+        [coin],
+        "auto" // Fee (auto-calculated or manual gas amount)
+    );
+    return result;
+}
+
+export const transferAction = {
+    name: "transfer",
+    description: "Transfer tokens between addresses on the same chain",
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        options: any,
+        callback?: HandlerCallback
+    ) => {
+        console.log("Transfer action handler called");
+
+        const chainName = runtime.getSetting("COSMOS_CHAIN_NAME");
+        const recoveryPhrase = runtime.getSetting("COSMOS_RECOVERY_PHRASE");
+
+        const { chainAssets, batch32prefix, feeToken, rpcEndpoint } =
+            await getChainSettings(chainName);
+
+        const wallet = await createWallet(recoveryPhrase, batch32prefix);
+
+        const [account] = await wallet.getAccounts();
+
+        const signingCosmWasmClient = await createSigningClient(
+            rpcEndpoint.address,
+            wallet,
+            feeToken.denom
+        );
+
+        const senderAddress = account.address;
+
+        // Compose transfer context
+        const transferContext = composeContext({
+            state,
+            template: transferTemplate,
+        });
+
+        // Generate transfer content
+        const content = await generateObjectDeprecated({
+            runtime,
+            context: transferContext,
+            modelClass: ModelClass.LARGE,
+        });
+
+        const transferContentValidator = z.object({
+            denomOrIbc: z.string(),
+            amount: z.string(),
+            toAddress: z.string(),
+        });
+
+        const transferContent = transferContentValidator.parse(content);
```

Comment:
i think we should wrap this code in a loop. If AI response not match zod schema, try again